src/modules/facility-investor/facility-investor.service.test.ts

```diff
@@ -0,0 +1,95 @@
+import { PROPERTIES } from '@ukef/constants';
+import { AcbsAuthenticationService } from '@ukef/modules/acbs/acbs-authentication.service';
+import { AcbsFacilityPartyService } from '@ukef/modules/acbs/acbs-facility-party.service';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { when } from 'jest-when';
+
+import { AcbsCreateFacilityPartyDto } from '../acbs/dto/acbs-create-facility-party.dto';
+import { DateStringTransformations } from '../date/date-string.transformations';
+import { FacilityInvestorService } from './facility-investor.service';
+
+describe('FacilityInvestorService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const idToken = valueGenerator.string();
+
+  let service: FacilityInvestorService;
+
+  let acbsFacilityPartyServiceCreatePartyForFacility: jest.Mock;
+
+  beforeEach(() => {
+    const acbsFacilityPartyService = new AcbsFacilityPartyService(null, null);
+    acbsFacilityPartyServiceCreatePartyForFacility = jest.fn();
+    acbsFacilityPartyService.createPartyForFacility = acbsFacilityPartyServiceCreatePartyForFacility;
+
+    const acbsAuthenticationService = new AcbsAuthenticationService(null, null, null);
+    const acbsAuthenticationServiceGetIdToken = jest.fn();
+    acbsAuthenticationService.getIdToken = acbsAuthenticationServiceGetIdToken;
+
+    service = new FacilityInvestorService(acbsAuthenticationService, acbsFacilityPartyService, dateStringTransformations);
+
+    when(acbsAuthenticationServiceGetIdToken).calledWith().mockResolvedValueOnce(idToken);
+  });
+
+  describe('createInvestorForFacility', () => {
+    const facilityIdentifier = valueGenerator.stringOfNumericCharacters();
+    const sectionIdentifier = PROPERTIES.FACILITY_INVESTOR.DEFAULT.sectionIdentifier;
+    const facilityStatusCode = PROPERTIES.FACILITY_INVESTOR.DEFAULT.facilityStatus.facilityStatusCode;
+    const involvedPartyIdentifier = PROPERTIES.FACILITY_INVESTOR.DEFAULT.involvedParty.partyIdentifier;
+    const effectiveDate = '9999-01-02';
```

Comment:
I can see `currency`, `effectiveDate` and `guaranteeExpiryDate` has been repeated three times, shall we pick these values from test constants please?

Reply:
Good idea, I've moved these to test constants